contracts/contracts/zapper/WOETHCCIPZapper.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IRouterClient } from "@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol";
+import { Client } from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol";
+import { IERC20 } from "@chainlink/contracts-ccip/src/v0.8/vendor/openzeppelin-solidity/v4.8.0/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@chainlink/contracts-ccip/src/v0.8/vendor/openzeppelin-solidity/v4.8.0/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC4626 } from "./../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { IOETHZapper } from "./../interfaces/IOETHZapper.sol";
+
+/**
+ * @title WOETH CCIP Zapper Contract
+ * @notice Hanldes logic to easily zap ETH on mainnet to WOETH on L2s like Arbitrum.
+ * @author Origin Protocol Inc
+ */
+
+contract WOETHCCIPZapper {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Event emitted when a zap occurs
+     * @param messageId Unique message identifier for each zap
+     * @param sender Address initiating the zap
+     * @param recipient Recipient address at destination chain
+     * @param amount Amount of ETH zapped
+     */
+    event Zap(
+        bytes32 indexed messageId,
+        address sender,
+        address recipient,
+        uint256 amount
+    );
+
+    /**
+     * @dev The destination chain selector
+     */
+    uint64 public immutable destinationChainSelector;
+
+    /**
+     * @dev The WOETH source chain (Mainnet)
+     */
+    IERC4626 public immutable woethSourceChain;
```

Comment:
The variable names here make these sound like they are chain ids rather than contract addresses? Perhaps `woethOnSourceChain`;